packages/drivers/odsp-driver/src/odspDocumentServiceFactoryCore.ts

```diff
@@ -80,6 +81,27 @@ export class OdspDocumentServiceFactoryCore implements IDocumentServiceFactory {
             throw new Error("File path should be provided!!");
         }
 
+        let fileInfo: INewFileInfo | IExistingFileInfo;
+        let createShareLinkParam: ShareLinkTypes | ISharingLinkKind | undefined;
+        if (odspResolvedUrl.fileName) {
+            createShareLinkParam = getSharingLinkParams(this.hostPolicy, searchParams);
+            fileInfo = {
+                type: 'New',
+                driveId: odspResolvedUrl.driveId,
+                siteUrl: odspResolvedUrl.siteUrl,
+                filePath,
+                filename: odspResolvedUrl.fileName,
+                createLinkType: createShareLinkParam,
+            };
+        } else if (odspResolvedUrl.itemId) {
```

Comment:
you should either convert this IF into unconditional ELSE branch (assuming there are only two options), or add ELSE branch at the end (assumes 3 options).